tests/unit/test_space.py

```diff
@@ -99,6 +122,51 @@ def test_discrete_search_space_sampling_raises_when_too_many_samples_are_request
         search_space.sample(num_samples)
 
 
+def test_discrete_search_space___mul___points_is_the_concatenation_of_original_points() -> None:
+    dss1 = DiscreteSearchSpace(tf.constant([[-1.0, -1.4], [-1.5, -3.6], [-0.5, -0.6]]))
+    dss2 = DiscreteSearchSpace(tf.constant([[1.0, 1.4], [1.5, 3.6]]))
+
+    product = dss1 * dss2
+
+    all_expected_points = tf.constant(
```

Comment:
You can use np.ndarray, and np.testing.assert_equal instead of custom `assert any(for ... in` loop.

Reply:
in most cases, yes, but here there's no guarantee of ordering so all we can say is that each point exists in the other tensor